components/markdown/src/snapshots/markdown__markdown__tests__def_before_use.snap

```diff
@@ -0,0 +1,55 @@
+---
```

Comment:
Can we actually do integration testing? Rather than testing the AST events, it would be better to compare the output html I think?

Reply:
Yep, sure. I can pass the vector of events to ` cmark::html::push_html()` and compare against generated html. Should I do it?

Reply:
@Keats ready for review.

Reply:
Sorry I've just realised: how does this interact with the new footnote feature from pulldown-cmark 0.10? It relies on it right?

Reply:
> Sorry I've just realised: how does this interact with the new footnote feature from pulldown-cmark 0.10? It relies on it right?

Well, not really.  The new feature affects the syntax of the footnotes, while my PR changes the parsed event stream.

As far as I understand, the new feature of `pulldown-cmark` is not required for this PR to work.

Reply:
Yes, they explicitly [state](https://github.com/pulldown-cmark/pulldown-cmark/blob/1cfaf02e516d24fd8c93f39b422548610c541aae/pulldown-cmark/examples/footnote-rewrite.rs#L17), that old footnote syntax is also supported.